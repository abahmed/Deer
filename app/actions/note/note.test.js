import configureStore from 'redux-mock-store'

// Actions to be tested
import * as noteActions from './index'

const mockStore = configureStore()
const store = mockStore()

describe('note_actions', () => {
  beforeEach(() => {
    store.clearActions()
  })

  describe('addNewNote', () => {
    test('Dispatches the correct action and payload', () => {
      store.dispatch(noteActions.addNewNote())
      expect(store.getActions()).toMatchSnapshot()
    })
  })

  describe('updateNotesList', () => {
    test('Dispatches the correct action and payload', () => {
      store.dispatch(noteActions.updateNotesList())
      expect(store.getActions()).toMatchSnapshot()
    })
  })

  describe('setSelectedNoteID', () => {
    test('Dispatches the correct action and payload', () => {
      store.dispatch(noteActions.setSelectedNoteID('a123'))
      expect(store.getActions()).toMatchSnapshot()
    })
  })

  describe('editSelectedNote', () => {
    test('Dispatches the correct action and payload', () => {
      store.dispatch(noteActions.editSelectedNote({
        title: '',
        content: '',
        modified: 1550849863632
      }))
      expect(store.getActions()).toMatchSnapshot()
    })
  })

  describe('updateSelectedNoteRev', () => {
    test('Dispatches the correct action and payload', () => {
      store.dispatch(noteActions.updateSelectedNoteRev('a123'))
      expect(store.getActions()).toMatchSnapshot()
    })
  })

  describe('deleteSelectedNote', () => {
    test('Dispatches the correct action and payload', () => {
      store.dispatch(noteActions.deleteSelectedNote())
      expect(store.getActions()).toMatchSnapshot()
    })
  })
  describe('updateSearchList', () => {
    test('Dispatches the correct action and payload', () => {
      store.dispatch(noteActions.updateSearchList())
      expect(store.getActions()).toMatchSnapshot()
    })
  })
})
