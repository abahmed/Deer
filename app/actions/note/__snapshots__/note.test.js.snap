// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`note_actions addNewNote Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "type": "ADD_NOTE",
  },
]
`;

exports[`note_actions deleteSelectedNote Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "type": "DELETE_SELECTED_NOTE",
  },
]
`;

exports[`note_actions editSelectedNote Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": Object {
      "content": "",
      "modified": 1550849863632,
      "title": "",
    },
    "type": "EDIT_SELECTED_NOTE",
  },
]
`;

exports[`note_actions setSelectedNoteID Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": "a123",
    "type": "SET_SELECTED_NOTE_ID",
  },
]
`;

exports[`note_actions updateNotesList Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "type": "NOTES_FETCH_SUCCESS",
  },
]
`;

exports[`note_actions updateSearchList Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "type": "NOTES_SEARCH_UPDATE",
  },
]
`;

exports[`note_actions updateSelectedNoteRev Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": "a123",
    "type": "UPDATE_SELECTED_NOTE_REV",
  },
]
`;
